azure/datalake/store/lib.py

```diff
@@ -157,12 +129,37 @@ def __init__(self, token):
     def signed_session(self):
         session = super(DataLakeCredential, self).signed_session()
         if time.time() - self.token['time'] > self.token['expiresIn'] - 100:
-            self.token = refresh_token(self.token)
+            self.refresh_token()
 
         scheme, token = self.token['tokenType'], self.token['access']
         header = "{} {}".format(scheme, token)
         session.headers['Authorization'] = header
         return session
+    
+    def refresh_token(self, authority=None):
+        """ Refresh an expired authorization token
+
+        Parameters
+        ----------
+        authority: string
+            The full URI of the authentication authority to authenticate against (such as https://login.microsoftonline.com/)
+        """
+        if self.token.get('refresh', False) is False:
```

Comment:
This was just a lift and shift of the existing method, but the reasoning is that we are trusting the calling class, in this case, only lib.auth() should ever construct one of these, but I am not sure how to prevent this class from being used by an end user. Basically, if there is no 'refresh' key in the token dictionary, then the token can't be refreshed. If there is a key, we will try to refresh the token with whatever the value is in that key. If the value is bad (or none or whatever), then refresh will fail at that point.